hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/HoodieParquetRealtimeInputFormat.java

```diff
@@ -80,6 +85,27 @@ protected HoodieDefaultTimeline filterInstantsTimeline(HoodieDefaultTimeline tim
   }
 
   void addProjectionToJobConf(final RealtimeSplit realtimeSplit, final JobConf jobConf) {
+
+    // set appropriate configs in jobConf to handle disabling of meta fields within canAddProjectionToJobConf and addRequiredProjectionFields
+    if (!realtimeSplit.getDeltaLogPaths().isEmpty()) {
+      // inject actual field info if meta fields are disabled
+      HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(jobConf).setBasePath(realtimeSplit.getBasePath()).build();
```

Comment:
this is an extra read of the `.hoodie`, from each record reader? This is a problem.  can we avoid this.